clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusStreamingUpdateAgent.java

```diff
@@ -236,255 +208,84 @@ public void executeAgent(Agent agent) {
+        if (tasks.containsKey(jobId)) {
+          tasks.get(jobId).add(task);
+        } else {
+          tasks.put(jobId, Stream.of(task).collect(Collectors.toCollection(HashSet::new)));
         }
-        cache.evictDeletedItems(SERVER_GROUPS.ns, Stream.of(serverGroupKey).collect(Collectors.toList()));
-        jobs.remove(job.getId());
-        log.info("Evicting job: {}, {} in {}", asgName, job.getId(), getAgentType());
-    }
-
-    private void updateTask(Job job, Task task, Map<String, Set<Task>> tasks) {
-      if (FILTERED_STATES.contains(task.getStatus().getState())) {
-        Map<String, Collection<CacheData>> cacheResults = new HashMap<>();
-        Map<String, CacheData> instancesCache = createCache();
-        Map<String, CacheData> serverGroupCache = createCache();
-        String asgName =
-          getAsgName(new com.netflix.spinnaker.clouddriver.titus.client.model.Job(job, Collections.EMPTY_LIST));
-        InstanceData instanceData =
-          new InstanceData(new com.netflix.spinnaker.clouddriver.titus.client.model.Task(task),
-            asgName,
-            account.getName(),
-            region.getName());
-        cacheInstance(instanceData, instancesCache);
-        CacheData serverGroupCacheData =
-          serverGroupCache.getOrDefault(instanceData.serverGroup, new MutableCacheData(instanceData.serverGroup));
-        Set<Task> taskIds = tasks.get(job.getId());
-        serverGroupCacheData.getRelationships()
-          .computeIfAbsent(INSTANCES.ns, key -> new HashSet<>())
-          .addAll(taskIds.stream()
-            .map(instanceTask -> (Keys.getInstanceV2Key(instanceTask.getId(), account.getName(), region.getName())))
-            .collect(Collectors.toSet()));
-        serverGroupCache.put(instanceData.serverGroup, serverGroupCacheData);
-        cacheResults.put(INSTANCES.ns, instancesCache.values());
-        cacheResults.put(SERVER_GROUPS.ns, serverGroupCache.values());
-        writeToCache(new DefaultCacheResult(cacheResults), false, TASK_TYPES);
-        log.info("Caching instance {} in {}", task.getId(), getAgentType());
+        changes++;
       } else if (FINISHED_TASK_STATES.contains(task.getStatus().getState())) {
-        tasks.get(job.getId()).remove(task);
-        cache.evictDeletedItems(INSTANCES.ns,
-          Stream.of(Keys.getInstanceV2Key(task.getId(), account.getName(), region.getName()))
-            .collect(Collectors.toList()));
-        log.info("Evicting task: {} in {}", task.getId(), getAgentType());
+        tasks.get(jobId).remove(task);
+        changes++;
       }
     }
 
-    private void writeToCache(CacheResult result, boolean evict, Collection<AgentDataType> providedTypes) {
-      Collection<String> authoritative = new HashSet<>(providedTypes.size());
-      for (AgentDataType type : providedTypes) {
-        if (type.getAuthority() == AgentDataType.Authority.AUTHORITATIVE) {
-          authoritative.add(type.getTypeName());
+    private void maybeWriteToCache() {
```

Comment:
Could you add a javadoc comment about when this would write and when it wouldn't?